app/models/session_activity.rb

```diff
@@ -1,22 +1,22 @@
 # frozen_string_literal: true
 
 class SessionActivity < ApplicationRecord
-  after_initialize :initialize_defaults
   SESSION_ACTIVITY_TYPES = %w[signup mhv dslogon idme mfa verify slo].freeze
 
   # Initial validations on creation
   validates :originating_request_id, presence: true
   validates :originating_ip_address, presence: true
-  validates :name, presence: true, inclusion: { in: SESSION_ACTIVITY_TYPES, allow_blank: true }
-  validates :status, presence: true, inclusion: { in: %w[incomplete success fail], allow_blank: true }
+  validates :name, presence: true, inclusion: { in: SESSION_ACTIVITY_TYPES }
+  validates :status, presence: true, inclusion: { in: %w[incomplete success fail] }
 
   # Additional validations on update
   # TODO: add these later.
 
-  private
+  def update_success
+    update(status: 'success')
+  end
```

Comment:
@lihanli I think we need to make sure we also update all of the attributes on the SessionActivity model based on information that we have: such as ip-address, user-agent (in the JSON blob). 

In the case of the failure we should probably track all of the above with the specific failure code. But I'll leave this up to @annaswims since VSP would have ownership of this now.